src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -61,22 +64,28 @@ public ObjectionService(ObjectionRepository objectionRepository,
                             ObjectionPatcher objectionPatcher,
                             FileTransferApiClient fileTransferApiClient,
                             ERICHeaderParser ericHeaderParser,
-                            ObjectionProcessor objectionProcessor) {
+                            ObjectionProcessor objectionProcessor,
+                            OracleQueryClient oracleQueryClient) {
         this.objectionRepository = objectionRepository;
         this.logger = logger;
         this.dateTimeSupplier = dateTimeSupplier;
         this.objectionPatcher = objectionPatcher;
         this.fileTransferApiClient = fileTransferApiClient;
         this.ericHeaderParser = ericHeaderParser;
         this.objectionProcessor = objectionProcessor;
+        this.oracleQueryClient = oracleQueryClient;
     }
 
     @Override
     public Objection createObjection(String requestId, String companyNumber, String ericUserId, String ericUserDetails) {
         Map<String, Object> logMap = buildLogMap(companyNumber, null, null);
         logger.infoContext(requestId, "Creating objection", logMap);
 
-        // TODO OBJ-231 process query result and return eligibility status
+        long actionCode = oracleQueryClient.getCompanyActionCode(companyNumber);
+
+        logger.debugContext(requestId, "Company action code is " + actionCode);
```

Comment:
Since the story doesn't specify how to deal with negative scenarios (and the fact that @markpit 's PR that will do the action code checks is close to being finished) I think just logging the code might be fine for now.